### PR TITLE
Fix Bit Overflow by limiting input to 32

### LINKED ISSUE #910: Input BitWidth on increasing over 32 does not work .
**Describe the bug**
The input when below 32 we can easily toggle the Bits from 0 to 1 or vice-versa. But over 32, let us assume 33 then the 33rd bit does not allows toggle , In this case you only can toggle the first 32 bits and rest remain static. If we do not want the circuit to accept input more than 32 bits then we should provide any alert otherwise we should fix this.
The spinner up button does not allows going beyond 32 but when user enters manually then this bug occurs.

**To Reproduce**
1. Go to 'circuitverse.org'
2. Click on 'simulator'
3. Scroll down to 'inputs' and select input , then enter a bitwidth greater than 32 manually.

**Expected behavior**
It should not create the input box for bitwidth over 32 as it does not responds to it.

**Screenshots**
Issue3

**Desktop (please complete the following information):**
 - OS: [Windows 10]
 - Browser [Chrome]

**Are you working on this issue? (Yes/No)**
No

### PR DESCRIPTION
**Pull request by YashKumarVerma**
Fixes #910 : Input fields can no longer have more than 32 width

#### Describe the changes you have made in this PR -
A function to automatically set widths greater than 32 to 32.

### Screenshots of the changes (If any) -
See the changes tab.

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

File: public/js/UX.js [hunk 1]
```diff
@@ -223,17 +223,17 @@ function showProperties(obj) {
         });
     }
 
-
-
-
-
-
+	function validateBitWidth(){
```
[the remark this answers is not in the document]
Comment (YashKumarVerma) [reply public/js/UX.js]: outdates, hence resolving.

File: public/js/UX.js [hunk 2]
```diff
@@ -223,17 +223,17 @@ function showProperties(obj) {
...
+		if($("[name='newBitWidth']").val() > 32){
```
[the remark this answers is not in the document]
Comment (YashKumarVerma) [reply public/js/UX.js]: outdates, hence resolving.

[the remark this answers is not in the document]
Comment (YashKumarVerma) [reply public/js/UX.js]: outdates, hence resolving.

[the remark this answers is not in the document]
Comment (YashKumarVerma) [reply public/js/UX.js]: outdates, hence resolving.

File: public/js/UX.js [hunk 3]
```diff
@@ -223,17 +223,17 @@ function showProperties(obj) {
...
+			$("[name='newBitWidth']").val(32);
```
[the remark this answers is not in the document]
Comment (YashKumarVerma) [reply public/js/UX.js]: outdates, hence resolving.

[the remark this answers is not in the document]
Comment (YashKumarVerma) [reply public/js/UX.js]: outdates, hence resolving.

File: public/js/UX.js [hunk 4]
```diff
@@ -223,17 +223,17 @@ function showProperties(obj) {
...
+		}
```
[the remark this answers is not in the document]
Comment (YashKumarVerma) [reply public/js/UX.js]: outdates, hence resolving.

File: public/js/UX.js [hunk 5]
```diff
@@ -223,17 +223,17 @@ function showProperties(obj) {
...
+            $("[name='newBitWidth']").val(32);
+        }
+    }
 
     $(".objectPropertyAttribute").on("change keyup paste click", function () {
         // return;
         //////console.log(this.name+":"+this.value);
 
-
+		validateBitWidth();
```
[the remark this answers is not in the document]
Comment (YashKumarVerma) [reply public/js/UX.js]: outdates, hence resolving.

[the remark this answers is not in the document]
Comment (YashKumarVerma) [reply public/js/UX.js]: outdates, hence resolving.

File: public/js/UX.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (YashKumarVerma) [reply public/js/UX.js]: outdates, hence resolving.

**Comment by YashKumarVerma** [comment]
> Working Well @YashKumarVerma  . But I am able to add a value like 32ee32 (which actually does not affects the output. After I leave the input and then selects it back , the bitwidth again goes back to normal i.e. 32) . Please fix this issue as only e , - , + , . , E can be entered in the textfield along with the numbers.
> Please have a look at the screenshot below :
> image

I am unable to understand the query raised here. Could you please elaborate?

**Comment by Shivansh2407** [comment]
I am able to input a value like 32+.eE which is not desired. The user should be restricted to enter only numeric value. Currently we can enter e,E,.,+,- and numbers. Please fix that too.

**Comment by YashKumarVerma** [comment]
> I am able to input a value like 32+.eE which is not desired. The user should be restricted to enter only numeric value. Currently we can enter e,E,.,+,- and numbers. Please fix that too.

thank you for the suggestions :+1

File: public/js/UX.js [hunk 6]
```diff
@@ -223,17 +223,25 @@ function showProperties(obj) {
...
+    function checkValidBitWidth() {
+        const selector = $("[name='newBitWidth']");
+        console.log(selector);
+        if (selector == undefined ||
+            selector.val() > 32 ||
+            selector.val() < 1 ||
+            !$.isNumeric(selector.val())) {
+            // fallback to previously saves state
+            selector.val(selector.attr("old-val"));
+        }else{
```
[the remark this answers is not in the document]
Comment (YashKumarVerma) [reply public/js/UX.js]: outdated, resolved in changes.

[the remark this answers is not in the document]
Comment (YashKumarVerma) [reply public/js/UX.js]: outdated, resolved in changes.

File: public/js/UX.js [hunk 7]
```diff
@@ -223,17 +223,25 @@ function showProperties(obj) {
...
-
-
-
-
+    function checkValidBitWidth() {
+        const selector = $("[name='newBitWidth']");
+        console.log(selector);
+        if (selector == undefined ||
+            selector.val() > 32 ||
+            selector.val() < 1 ||
```
[the remark this answers is not in the document]
Comment (YashKumarVerma) [reply public/js/UX.js]: outdated, resolved in changes.

File: public/js/UX.js [hunk 8]
```diff
@@ -223,17 +223,25 @@ function showProperties(obj) {
...
-
-
-
-
-
+    function checkValidBitWidth() {
+        const selector = $("[name='newBitWidth']");
+        console.log(selector);
+        if (selector == undefined ||
+            selector.val() > 32 ||
```
[the remark this answers is not in the document]
Comment (YashKumarVerma) [reply public/js/UX.js]: outdated, resolved in changes.

File: public/js/UX.js [hunk 9]
```diff
@@ -223,17 +223,25 @@ function showProperties(obj) {
...
 
-
-
-
-
-
+    function checkValidBitWidth() {
+        const selector = $("[name='newBitWidth']");
+        console.log(selector);
+        if (selector == undefined ||
```
[the remark this answers is not in the document]
Comment (YashKumarVerma) [reply public/js/UX.js]: outdated, resolved in changes.

File: public/js/UX.js [hunk 10]
```diff
@@ -223,17 +223,25 @@ function showProperties(obj) {
...
     }
 
-
-
-
-
-
+    function checkValidBitWidth() {
+        const selector = $("[name='newBitWidth']");
+        console.log(selector);
```
[the remark this answers is not in the document]
Comment (YashKumarVerma) [reply public/js/UX.js]: outdated, resolved in changes.

**Comment by satu0king** [comment]
Very smart!

**Pull request merged**